internal/route/setup.go

```diff
@@ -1,26 +1,38 @@
 package route
 
 import (
-	"github.com/dilyara4949/employees-api/internal/config"
+	conf "github.com/dilyara4949/employees-api/internal/config"
 	"net/http"
 
 	"github.com/dilyara4949/employees-api/internal/controller"
 	"github.com/dilyara4949/employees-api/internal/middleware"
 )
 
-func SetUpRouter(employeesController *controller.EmployeesController, positionsController *controller.PositionsController, config config.Config, mux *http.ServeMux) {
+func SetUpRouter(employeesController *controller.EmployeesController, positionsController *controller.PositionsController, config conf.Config, mux *http.ServeMux) {
 
-	auth := middleware.NewJWTAuth(config.JWTTokenSecret)
+	//auth := middleware.NewJWTAuth(config.JWTTokenSecret)
 
-	mux.HandleFunc("GET /positions/{id}", auth.Auth(positionsController.GetPosition))
-	mux.HandleFunc("POST /positions", auth.Auth(positionsController.CreatePosition))
-	mux.HandleFunc("DELETE /positions/{id}", auth.Auth(positionsController.DeletePosition))
-	mux.HandleFunc("PUT /positions/{id}", auth.Auth(positionsController.UpdatePosition))
-	mux.HandleFunc("GET /positions", auth.Auth(positionsController.GetAllPositions))
+	mux.HandleFunc("GET /positions/{id}", LogCorrelationIDTimer(positionsController.GetPosition, config.JWTTokenSecret))
+	mux.HandleFunc("POST /positions", LogCorrelationIDTimer(positionsController.CreatePosition, config.JWTTokenSecret))
+	mux.HandleFunc("DELETE /positions/{id}", LogCorrelationIDTimer(positionsController.DeletePosition, config.JWTTokenSecret))
+	mux.HandleFunc("PUT /positions/{id}", LogCorrelationIDTimer(positionsController.UpdatePosition, config.JWTTokenSecret))
+	mux.HandleFunc("GET /positions", LogCorrelationIDTimer(positionsController.GetAllPositions, config.JWTTokenSecret))
 
-	mux.HandleFunc("GET /employees/{id}", auth.Auth(employeesController.GetEmployee))
-	mux.HandleFunc("POST /employees", auth.Auth(employeesController.CreateEmployee))
-	mux.HandleFunc("DELETE /employees/{id}", auth.Auth(employeesController.DeleteEmployee))
-	mux.HandleFunc("PUT /employees/{id}", auth.Auth(employeesController.UpdateEmployee))
-	mux.HandleFunc("GET /employees", auth.Auth(employeesController.GetAllEmployees))
+	mux.HandleFunc("GET /employees/{id}", LogCorrelationIDTimer(employeesController.GetEmployee, config.JWTTokenSecret))
+	mux.HandleFunc("POST /employees", LogCorrelationIDTimer(employeesController.CreateEmployee, config.JWTTokenSecret))
+	mux.HandleFunc("DELETE /employees/{id}", LogCorrelationIDTimer(employeesController.DeleteEmployee, config.JWTTokenSecret))
+	mux.HandleFunc("PUT /employees/{id}", LogCorrelationIDTimer(employeesController.UpdateEmployee, config.JWTTokenSecret))
+	mux.HandleFunc("GET /employees", LogCorrelationIDTimer(employeesController.GetAllEmployees, config.JWTTokenSecret))
+}
+
+func LogCorrelationIDTimer(endpoint http.HandlerFunc, JWTTokenSecret string) http.HandlerFunc {
```

Comment:
No need for this function to be exported (name starts with capital letter), IMO, because it will be used only here.

Reply:
Fixed